test/hotspot/jtreg/compiler/vectorization/TestRoundVectorDoubleRandom.java

```diff
@@ -0,0 +1,156 @@
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/**
+ * @test
+ * @key randomness
+ * @bug 8321011
+ * @summary Test vector intrinsic for Math.round(double) in full 64 bits range.
+ *
+ * @library /test/lib /
+ * @run main compiler.vectorization.TestRoundVectorDoubleRandom
+ */
+
+package compiler.vectorization;
+
+import java.util.Random;
+import compiler.lib.ir_framework.DontCompile;
+import compiler.lib.ir_framework.IR;
+import compiler.lib.ir_framework.IRNode;
+import compiler.lib.ir_framework.Run;
+import compiler.lib.ir_framework.RunInfo;
+import compiler.lib.ir_framework.Test;
+import compiler.lib.ir_framework.TestFramework;
+import compiler.lib.ir_framework.Warmup;
+
+public class TestRoundVectorDoubleRandom {
+  private static final int ITERS  = 11000;
+  private static final int ARRLEN = 997;
+  private static final double ADD_INIT = -7500.;
+
+  private static final double[] input = new double[ARRLEN];
+  private static final long [] res = new long[ARRLEN];
+  private static final Random rand = new Random();
+
+  public static void main(String args[]) {
+    TestFramework.runWithFlags("-XX:-TieredCompilation", "-XX:CompileThresholdScaling=0.3", "-XX:MaxVectorSize=16");
+    TestFramework.runWithFlags("-XX:-TieredCompilation", "-XX:CompileThresholdScaling=0.3", "-XX:MaxVectorSize=32");
+  }
+
+  @DontCompile
+  long golden_round(double d) {
+    return Math.round(d);
+  }
+
+  @Test
+  @IR(counts = {IRNode.ROUND_VD, "> 0"},
+      applyIfPlatform = {"x64", "true"},
+      applyIfCPUFeature = {"avx512dq", "true"})
+  @IR(counts = {IRNode.ROUND_VD, "> 0"},
+      applyIfPlatform = {"aarch64", "true"})
+  void test_round(long[] a0, double[] a1) {
+    for (int i = 0; i < a0.length; i+=1) {
+      a0[i] = Math.round(a1[i]);
+    }
+  }
+
+  @Run(test = "test_round")
+  @Warmup(ITERS)
+  void test_rounds(RunInfo runInfo) {
+    // Initialize
+    for (int i = 0; i < ARRLEN; i++) {
+      double val = ADD_INIT+(double)i;
+      input[i] = val;
+    }
+
+    test_round(res, input);
+    // skip test/verify when warming up
+    if (runInfo.isWarmUp()) {
+      return;
+    }
+
+    int errn = 0;
+    // a double precise float point is composed of 3 parts: sign/e(exponent)/f(signicant)
+    // e (exponent) part of a float value
+    final int eShift = 52;
+    final int eWidth = 11;
+    final int eBound = 1 << eWidth;
+    // f (significant) part of a float value
+    final int fWidth = eShift;
+    final long fBound = 1 << fWidth;
+    final int fNum = 256;
+
+    // prepare for data of f (i.e. significand part)
+    long fis[] = new long[fNum];
+    int fidx = 0;
+    for (; fidx < fWidth; fidx++) {
+      fis[fidx] = 1 << fidx;
```

Comment:
```suggestion
      fis[fidx] = 1L << fidx;
```
One more here, please be careful with this!